### PR TITLE
Subscription API

### PR DESCRIPTION
**Pull request by tamslo**

### CODE REVIEW

File: src/services/updateSubscription.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+const updateSubscriptionInDb = require('../queries/updateSubscription');
+const deleteSubscription = require('../queries/deleteSubscription');
+
+function updateSubscription(subscription, subscriptionId) {
+    return new Promise((resolve, reject) => {
+        const { icsUrl, description, scopeIds } = subscription;
+        // since per posted scopeId one event is returned, given the id the
```
Comment (MrSerth) [inline src/services/updateSubscription.js]: huch, I don't get it. Why is there one event (which one?) returned per posted scopeID?

File: src/routes/subscriptions.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+                );
+            });
+            returnSuccess(res, 200, insertedSubscriptions);
+        })
+        .catch((error) => returnError(error));
+});
+
+router.put('/subscriptions/:subscriptionId', authorize, jsonApiToJson, function (req, res) {
+    const subscriptionId = req.params.subscriptionId;
+    // we only allow the subscription belonging to the id to be edited
```
Comment (MrSerth) [inline src/routes/subscriptions.js]: I know what you mean here, but I would prefer to change that comment. What about "We only allow one subscription (belonging to the given id) to be edited"?

File: docs/schulcloud_calendar_API.yaml [hunk 3]
```diff
@@ -377,19 +375,19 @@ paths:
...
     parameters:
-      - name: feed-id
+      - name: subscription-id
         in: path
         required: true
         type: string
     put:
       summary: Update synchronisation settings for a subscribed feed (X).
-      description: The calendar service will delete the old external feed subscription completely. Therefore, you MUST ensure to re-submit the subscription with all details.
+      description: The calendar service will delete the old external feed subscriptions completely. Therefore, you MUST ensure to re-submit the subscriptions with all details.
```
Comment (tamslo) [inline docs/schulcloud_calendar_API.yaml]: You are right, should be singular, will change

File: src/queries/deleteSubscription.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+const client = require('../infrastructure/database');
+const errorMessage = require('./utils/errorMessage');
+
+function deleteSubscription(subscriptionId) {
+    return new Promise(function(resolve, reject) {
+        const query = 'DELETE FROM feeds WHERE id = $1 RETURNING *';
```
Comment (tamslo) [inline src/queries/deleteSubscription.js]: :+1:

File: src/queries/getSubscriptions.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    query = `${query} ORDER BY id ASC;`;
+    return { query, params };
+}
+
+function noParamsGiven(filter) {
+    const { scopeId, subscriptionId, lastUpdateFailed } = filter;
+    return !scopeId
+        && !subscriptionId
+        // the check when lastUpdateFailed is false fails intentionally since
+        // we assume that it is not a useful query
```
Comment (tamslo) [inline src/queries/getSubscriptions.js]: Set to true it is, I meant set to false it is not really useful

File: src/routes/subscriptions.js (hunk 2 above)
Comment (tamslo) [inline src/routes/subscriptions.js]: Will do

File: src/services/updateSubscription.js (hunk 1 above)
Comment (tamslo) [inline src/services/updateSubscription.js]: You're right, event should be subscribtion. Then everything clear?

File: src/queries/getSubscriptions.js (hunk 5 above)
Comment (MrSerth) [inline src/queries/getSubscriptions.js]: okay, sure.

File: src/services/updateSubscription.js (hunk 1 above)
Comment (MrSerth) [inline src/services/updateSubscription.js]: Yes. We should add a note e.g. to Swagger that updating many subscriptions, created by a single POST and separateUsers = true, require many PUTs to update all and that there is no bulk update for separated subscriptions.

**Pull request merged**